### ISSUE TITLE
Codecov website not merging duplicate coverage reports for a single commit

### ISSUE THREAD

**Issue by bstaletic** [codecov/codecov-action#1083]
In case this is the wrong place to file a bug report, please point me in the right direction.

Yesterday I was investigating why codecov was not commenting on ycm-core/ycmd pull requests and found codecov/feedback#126 to be the culprit.
https://github.com/ycm-core/ycmd/pull/1708 was used for experimenting.

Yesterday codecov action managed to upload 5 out of 7 reports. Today I reran the entire CI for that pull request and... to my surprise, codecov created a comment only after two more. Then updated the comment after the rest of the CI finished.

Looking at [codecov.org](https://app.codecov.io/gh/ycm-core/ycmd/commit/46430e0731bcf447ab086e33fd565ebd0350de60) it says there are 12 reports for that commit. That's 5 more than expected.

**Comment by thomasrockhu-codecov** [comment]
@bstaletic apologies this is a bit old now. The solution was to add the Codecov token likely. I'm going to close due to age, but if you are still experiencing an issue, please open another issue in our [feedback repository](https://github.com/codecov/feedback)
